nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/dynamodb/PutDynamoDB.java

```diff
@@ -0,0 +1,182 @@
+    @WritesAttribute(attribute = AbstractDynamoDBProcessor.DYNAMODB_ERROR_EXCEPTION_MESSAGE, description = "Dynamo db exception message"),
+    @WritesAttribute(attribute = AbstractDynamoDBProcessor.DYNAMODB_ERROR_CODE, description = "Dynamo db error code"),
+    @WritesAttribute(attribute = AbstractDynamoDBProcessor.DYNAMODB_ERROR_MESSAGE, description = "Dynamo db error message"),
+    @WritesAttribute(attribute = AbstractDynamoDBProcessor.DYNAMODB_ERROR_TYPE, description = "Dynamo db error type"),
+    @WritesAttribute(attribute = AbstractDynamoDBProcessor.DYNAMODB_ERROR_SERVICE, description = "Dynamo db error service"),
+    @WritesAttribute(attribute = AbstractDynamoDBProcessor.DYNAMODB_ERROR_RETRYABLE, description = "Dynamo db error is retryable"),
+    @WritesAttribute(attribute = AbstractDynamoDBProcessor.DYNAMODB_ERROR_REQUEST_ID, description = "Dynamo db error request id"),
+    @WritesAttribute(attribute = AbstractDynamoDBProcessor.DYNAMODB_ERROR_STATUS_CODE, description = "Dynamo db error status code"),
+    @WritesAttribute(attribute = AbstractDynamoDBProcessor.DYNAMODB_ITEM_IO_ERROR, description = "IO exception message on creating item")
+    })
+@ReadsAttributes({
+    @ReadsAttribute(attribute = AbstractDynamoDBProcessor.DYNAMODB_ITEM_HASH_KEY_VALUE, description = "Items hash key value" ),
+    @ReadsAttribute(attribute = AbstractDynamoDBProcessor.DYNAMODB_ITEM_RANGE_KEY_VALUE, description = "Items range key value" )
+    })
+public class PutDynamoDB extends AbstractWriteDynamoDBProcessor {
+
+   public static final List<PropertyDescriptor> properties = Collections.unmodifiableList(
+            Arrays.asList(TABLE, HASH_KEY_NAME, RANGE_KEY_NAME, HASH_KEY_VALUE, RANGE_KEY_VALUE,
+                HASH_KEY_VALUE_TYPE, RANGE_KEY_VALUE_TYPE, JSON_DOCUMENT, DOCUMENT_CHARSET, BATCH_SIZE,
+                REGION, ACCESS_KEY, SECRET_KEY, CREDENTIALS_FILE, AWS_CREDENTIALS_PROVIDER_SERVICE, TIMEOUT, SSL_CONTEXT_SERVICE));
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) {
+        List<FlowFile> flowFiles = session.get(context.getProperty(BATCH_SIZE).asInteger());
+        if (flowFiles == null || flowFiles.size() == 0) {
+            return;
+        }
+
+        Map<ItemKeys,FlowFile> keysToFlowFileMap = new HashMap<>();
+
+        final String table = context.getProperty(TABLE).getValue();
+
+        final String hashKeyName = context.getProperty(HASH_KEY_NAME).getValue();
+        final String hashKeyValueType = context.getProperty(HASH_KEY_VALUE_TYPE).getValue();
+        final String rangeKeyName = context.getProperty(RANGE_KEY_NAME).getValue();
+        final String rangeKeyValueType = context.getProperty(RANGE_KEY_VALUE_TYPE).getValue();
+        final String jsonDocument = context.getProperty(JSON_DOCUMENT).getValue();
```

Comment:
Hey @Paul424!

Sorry for missing your previous question, my filters consumed the notification email, but saw that you got a resolution and your approach sounds great.

The key thing to underscore is that a FlowFile is a set of attributes and a content section.  Expression Language (EL) is a way to do some quick operations on those attributes.  Attributes are effectively metadata about the content and aid in making decisions and characterizing the data upfront so that the content can make its way through the appropriate pipeline and processing. 

If you happen to run into anything else in terms of AWS processors, or even NiFi in general, feel free to reach out to the community.  (http://nifi.apache.org/mailing_lists.html)  Any feedback on how we could make this or anything else easier is always greatly appreciated!